pkg/ring/ring.go

```diff
@@ -377,3 +378,66 @@ func (r *Ring) Collect(ch chan<- prometheus.Metric) {
 		r.name,
 	)
 }
+
+// Subring returns a ring of n ingesters from the given ring
+// Subrings are meant only for ingestor lookup and should have their data externalized.
+func (r *Ring) Subring(key uint32, n int) (ReadRing, error) {
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+	if r.ringDesc == nil || len(r.ringTokens) == 0 || n <= 0 {
+		return nil, ErrEmptyRing
+	}
+
+	if n > len(r.ringDesc.Ingesters) {
```

Comment:
While doing some tests I've tried to intentionally hit this. I'm wondering if this is a real problem for which returning error is desired. If the subring is larger than the current number of ingesters, it means the returned subring will include all ingesters, which is not technically wrong. I do understand it's not desiderable (why do you use the subring, then?) but it could actually happen in case of an incident where you find yourself forgetting many ingesters while spinning up new ones.

What if we just remove this line (and cover this case with a test)?

Reply:
Good idea.

Modified that line to set n to number of ingesters, and modified test accordingly.